dashboard/test/helpers/gdpr_dialog_helper_test.rb

```diff
@@ -0,0 +1,33 @@
+require 'test_helper'
+
+class GdprDialogHelperTest < ActionView::TestCase
+  setup do
+    @user = build :user
+
+    @request = ActionDispatch::Request.new({})
+    @request.stubs(:gdpr?).returns(false)
+    @request.stubs(:params).returns({})
+  end
+
+  test 'do not show GDPR dialog if no conditions met' do
+    refute GdprDialogHelper.show?(@user, @request)
+  end
+
+  test 'do not show GDPR dialog if user has accepted data transfer agreement' do
+    @user.data_transfer_agreement_accepted = true
```

Comment:
Would appreciate help understanding use of instance variables in tests...I think I may not be doing this correctly, in that I'm making state changes to instance variables that might get leaked between tests?

Reply:
I think your instincts are right here!

I'd recommend one of two approaches. The first option is what you've done in the ThankDonorsInterstitialHelperTest test beloy; you write this test basically as you've written it, with a user object shared between all tests, but then for any tests which require you to modify the instance variable, create a new one just for that test rather than using the shared. For example:
```suggestion
    accepted_user = build :user
    accepted_user.data_transfer_agreement_accepted = true
```


The second option would be to identify up front which variations you'll need, and then create a shared instance variable for each. For example:

```ruby

  setup do
    @accepted_user = build :user, data_transfer_agreement_accepted: true
    @default_user = build :user

    @request = ActionDispatch::Request.new({})
    @request.stubs(:gdpr?).returns(false)
    @request.stubs(:params).returns({})
  end
```